ecs-agent/netlib/platform/common_linux.go

```diff
@@ -0,0 +1,200 @@
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to translate network configuration")
+		}
+	case ecs.NetworkModeBridge:
+		return nil, errors.New("not implemented")
+	case ecs.NetworkModeHost:
+		return nil, errors.New("not implemented")
+	case ecs.NetworkModeNone:
+		return nil, errors.New("not implemented")
+	default:
+		return nil, errors.New("invalid network mode: " + mode)
+	}
+
+	return &tasknetworkconfig.TaskNetworkConfig{
+		NetworkNamespaces: netNSs,
+		NetworkMode:       mode,
+	}, nil
+}
+
+func (c *common) GetNetNSPath(netNSName string) string {
+	return c.nsUtil.GetNetNSPath(netNSName)
+}
+
+// buildAWSVPCNetworkNamespaces returns list of NetworkNamespace which will be used to
+// create the task's network configuration. All cases except those for FoF is covered by
+// this method. FoF requires a separate specific implementation because the network setup
+// is different due to the presence of the microVM.
+// Use cases covered by this method are:
+//  1. Single interface, network namespace (the only externally available config).
+//  2. Single netns, multiple interfaces (For a non-managed multi-ENI experience. Eg EKS use case).
+//  3. Multiple netns, multiple interfaces (future use case for internal customer who need
+//     a managed multi-ENI experience).
+func (c *common) buildAWSVPCNetworkNamespaces(taskID string,
+	taskPayload *ecsacs.Task) ([]*tasknetworkconfig.NetworkNamespace, error) {
+	if len(taskPayload.ElasticNetworkInterfaces) == 0 {
+		return nil, errors.New("interfaces list cannot be empty")
+	}
+	// If task payload has only one interface, the network configuration is
+	// straight forward. It will have only one network namespace containing
+	// the corresponding network interface.
+	// Empty Name fields in network interface names indicate that all
+	// interfaces share the same network namespace. This use case is
+	// utilized by certain internal teams like EKS on Fargate.
+	if len(taskPayload.ElasticNetworkInterfaces) == 1 ||
+		aws.StringValue(taskPayload.ElasticNetworkInterfaces[0].Name) == "" {
+		primaryNetNS, err := c.buildSingleNSNetConfig(taskID,
+			0,
+			taskPayload.ElasticNetworkInterfaces,
+			taskPayload.ProxyConfiguration)
+		if err != nil {
+			return nil, err
+		}
+
+		return []*tasknetworkconfig.NetworkNamespace{primaryNetNS}, nil
+	}
+
+	// Create a map for easier lookup of ENIs by their names.
+	ifNameMap := make(map[string]*ecsacs.ElasticNetworkInterface, len(taskPayload.ElasticNetworkInterfaces))
+	for _, iface := range taskPayload.ElasticNetworkInterfaces {
+		ifNameMap[networkinterface.GetInterfaceName(iface)] = iface
+	}
+
+	// Proxy configuration is not supported yet in a multi-ENI / multi-NetNS task.
+	if taskPayload.ProxyConfiguration != nil {
+		return nil, errors.New("unexpected proxy config found")
+	}
+
+	// The number of network namespaces required to create depends on the
+	// number of unique interface names list across all container definitions
```

Comment:
I believe device names are unique instance-wide at a single given time, but are not guaranteed to be unique over time.  if we use the device name as our identifier we'll run the risk of collision.  
So for handling a single task payload, device name is fine for differentiation, but we'll want to reference the [NetworkInterfaceId](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-networkinterfaceattachment.html#cfn-ec2-networkinterfaceattachment-networkinterfaceid) where possible to guarantee uniqueness over time.

Reply:
I will reword this comment here. When I mentioned `interface names`, I wasn't referring to the device name on the host. I was referring to `ElasticNetworkInterface.Name` in the task payload. This `name` field is configurable by the customer in the task definition. If the customer choses to not configure it, we default it to a string derived from the ENI's mac address.